From 0cfc29a78d9e1c07e115d08f86814ed53052ef61 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Mon, 2 Sep 2019 18:56:37 +0300
Subject: [PATCH 3/3] pci: spr2803: hack pcie resources allocation

spr2803 declares class id = 0x0, where the kernel ignores this type of
class and does not allocate PCIe resources to it (BARs).

This hack removes the kernel check and does allocate resources to it.

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 drivers/pci/setup-bus.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/setup-bus.c b/drivers/pci/setup-bus.c
index 958da7db9033..0b18b70f0e86 100644
--- a/drivers/pci/setup-bus.c
+++ b/drivers/pci/setup-bus.c
@@ -187,7 +187,7 @@ static void __dev_sort_resources(struct pci_dev *dev,
 	u16 class = dev->class >> 8;
 
 	/* Don't touch classless devices or host bridges or ioapics.  */
-	if (class == PCI_CLASS_NOT_DEFINED || class == PCI_CLASS_BRIDGE_HOST)
+	if (/*class == PCI_CLASS_NOT_DEFINED ||*/ class == PCI_CLASS_BRIDGE_HOST)
 		return;
 
 	/* Don't touch ioapic devices already enabled by firmware */
-- 
2.17.1

